test/factories/delegatedManagerFactory.spec.ts

```diff
@@ -0,0 +1,707 @@
+      await subject();
+
+      const initializeParams = await delegatedManagerFactory.initializeState(subjectSetToken);
+      const delegatedManager = await deployer.manager.getDelegatedManager(initializeParams.manager);
+
+      expect(await delegatedManager.setToken()).eq(subjectSetToken);
+      expect(await delegatedManager.factory()).eq(delegatedManagerFactory.address);
+      expect(await delegatedManager.methodologist()).eq(subjectMethodologist);
+      expect(await delegatedManager.useAssetAllowlist()).eq(true);
+    });
+
+    it("should set the intialization state correctly", async() => {
+      const newManagerAddress = await delegatedManagerFactory.callStatic.createManager(
+        subjectSetToken,
+        subjectOwner,
+        subjectMethodologist,
+        subjectOperators,
+        subjectAssets,
+        subjectExtensions
+      );
+
+      await subject();
+
+      const initializeParams = await delegatedManagerFactory.initializeState(subjectSetToken);
+
+      expect(initializeParams.deployer).eq(owner.address);
+      expect(initializeParams.owner).eq(subjectOwner);
+      expect(initializeParams.isPending).eq(true);
+      expect(initializeParams.manager).eq(newManagerAddress);
+    });
+
+    it("should emit a DelegatedManagerDeployed event", async() => {
+      const managerAddress = await delegatedManagerFactory.callStatic.createManager(
+        subjectSetToken,
+        subjectOwner,
+        subjectMethodologist,
+        subjectOperators,
+        subjectAssets,
+        subjectExtensions
+      );
+
+      await expect(subject()).to.emit(delegatedManagerFactory, "DelegatedManagerCreated").withArgs(
+        subjectSetToken,
+        managerAddress,
+        owner.address
+      );
+    });
+
+    describe("when the caller is not the SetToken manager", async () => {
+      beforeEach(() => {
+        subjectCaller = otherAccount;
+      });
+
+      it("should revert", async () => {
+        await expect(subject()).to.be.revertedWith("Must be manager");
+      });
+    });
+
+    describe("when the assets array is non-empty but missing some component elements", async() => {
+      beforeEach(async() => {
+        subjectAssets = [setV2Setup.wbtc.address];
+      });
+
+      it("should revert", async() => {
+        await expect(subject()).to.be.revertedWith("Asset list must include all components");
+      });
+    });
+
+    // This test exists to cover the condition on the branch which validates components are assets
+    // Just verifying that creation state is pending
+    describe("when the assets array is empty", async() => {
+      beforeEach(() => {
+        subjectAssets = [];
+      });
+
+      it("should set the intialization state correctly", async() => {
+        await subject();
+
+        const initializeParams = await delegatedManagerFactory.initializeState(subjectSetToken);
+
+        expect(initializeParams.isPending).eq(true);
+      });
```

Comment:
Add test to verify that DelegatedManager''s `useAssetAllowlist` is false

Reply:
[1901b6a](https://github.com/SetProtocol/set-v2-strategies/pull/14/commits/1901b6ac4ebe9007336d6b8621ba35ea424477fb)